lib/sinon/sandbox.js

```diff
@@ -278,6 +285,33 @@ function Sandbox() {
         return replacement;
     };
 
+    sandbox.define = function define(object, property, value) {
+        const descriptor = getPropertyDescriptor(object, property);
+
+        if (typeof descriptor !== "undefined") {
+            throw new TypeError(
+                `Cannot define the already existing property ${valueToString(
+                    property
+                )}. Perhaps you meant sandbox.replace()?`
+            );
+        }
+
+        if (typeof value === "undefined") {
+            throw new TypeError("Expected value argument to be defined");
+        }
+
+        // Check just in case we allow to replace properties by undefined in the future.
+        // In the implementation at the moment of writing, this check always succeeds.
```

Comment:
I meant there's no way the value was replaced before, because we already checked that the descriptor is still undefined. I removed the comment, now I don't think it's really useful.